external/sql/storm-sql-core/src/jvm/org/apache/storm/sql/compiler/backends/trident/TridentPostOrderRelNodeVisitor.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.sql.compiler.backends.trident;
+
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.core.Aggregate;
+import org.apache.calcite.rel.core.Calc;
+import org.apache.calcite.rel.core.Collect;
+import org.apache.calcite.rel.core.Correlate;
+import org.apache.calcite.rel.core.Exchange;
+import org.apache.calcite.rel.core.Filter;
+import org.apache.calcite.rel.core.Join;
+import org.apache.calcite.rel.core.Project;
+import org.apache.calcite.rel.core.Sample;
+import org.apache.calcite.rel.core.Sort;
+import org.apache.calcite.rel.core.TableModify;
+import org.apache.calcite.rel.core.TableScan;
+import org.apache.calcite.rel.core.Uncollect;
+import org.apache.calcite.rel.core.Window;
+import org.apache.calcite.rel.stream.Delta;
+import org.apache.storm.trident.TridentTopology;
+import org.apache.storm.trident.fluent.IAggregatableStream;
+import org.apache.storm.trident.topology.TridentTopologyBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class TridentPostOrderRelNodeVisitor {
```

Comment:
I just wanted to avoid any changes on standalone mode since we're constructing topology (or components) differently for two modes now.
And also I'm curious that Trident stream building pattern is common for others.

What do you think if we change PostOrderRelNodeVisitor to below?

```
public abstract class PostOrderRelNodeVisitor<T> {
  public final T traverse(RelNode n) throws Exception {
    List<T> inputStreams = new ArrayList<>();
    for (RelNode input : n.getInputs()) {
      inputStreams.add(traverse(input));
    }

    if (n instanceof Aggregate) {
      return visitAggregate((Aggregate) n, inputStreams);
    ...
  }

  public T visitAggregate(Aggregate aggregate, List<T> inputStreams) throws Exception {
    return defaultValue(aggregate, inputStreams);
  }
...
```

Is it OK for you to apply this for standalone mode?